src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -201,11 +202,14 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
           case wf if !wf.status.isDone => Future { executionServiceDAO.abort(wf.workflowId, userInfo) }
         }
 
-        val accessGroups = workspaceContext.workspace.accessLevels.values.map(containerDAO.authDAO.loadGroup(_, txn)).collect { case Some(g) => g }
-        gcsDAO.deleteWorkspace(workspaceContext.workspace.bucketName, accessGroups.toSeq, bucketDeletionMonitor).map { _ =>
-          containerDAO.authDAO.deleteWorkspaceAccessGroups(workspaceContext.workspace, txn)
-          containerDAO.workspaceDAO.delete(workspaceName, txn)
+        // use a set because the two maps may be duplicates
+        val groupRefs: Set[RawlsGroupRef] = workspaceContext.workspace.accessLevels.values.toSet ++ workspaceContext.workspace.realmACLs.values
+        val groups = groupRefs.map(containerDAO.authDAO.loadGroup(_, txn)).collect { case Some(g) => g }
 
+        Future.traverse(groups) { gcsDAO.deleteGoogleGroup } map { _ =>
```

Comment:
You've used map here and foreach below. Any reason why? Do we go with the rule of "use map when you return results and foreach when you don't"?